### DISCUSSION TITLE
Welcome to idaes-pse Discussions!

### DISCUSSION THREAD

**Discussion by dangunter** [IDAES/idaes-pse#116] [General]
## Welcome!
  We’re using Discussions as a place to connect with stakeholders and other members of our community. Please use it as a place to:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  
All are welcome.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by alguzur** [comment]
Hello Guys,
This is Alex Guzman from Tohoku Univ. Japan.
Currently, I am Simulating a BFB Reactor (By the way great tool!!). However, I am struggling with the understanding of some parameters of the model. Such as the "available volume" of the solid. May you share with me more information about this property and if possible the source employed for the example:
(idaes-pse/idaes/gas_solid_contactors/properties/methane_iron_OC_reduction/hetero_reactions.py)
Thanks!!

**Reply by andrewlee94** [reply]
Hi @alguzur,

Thank you for taking the time to look at IDAES. I've reached out to the developer of the BFB reactor model to ask them if they could answer your questions.

**Reply by chineduobiora** [reply]
Hi @alguzur
 
The available volume for reaction is the amount of reaction sites on the oxygen carrier per the volume of the oxygen carrier. It is oxygen carrier dependent and will vary with the amount of inert/impurities in the OC, and the porosity of the oxygen carrier. The number should vary throughout the reaction because the inert mass fraction and oxygen carrier porosity change with reaction.
However, an average value is used here for simplification purposes. It is averaged over the complete cycle of the OC i.e., from a fully oxidized state to a fully reduced state. The number in the reaction property package wasn’t taken from a standard reference but from some unpublished in-house work/calculations.
You can approximately calculate this value for each temporal and spatial position in the model by using the equation (1 – properties[t,x].mass_frac_comp[‘Al2O3’])*(1 – properties[t,x].particle_porosity).

I hope this helps.

**Comment by jkleinekorte** [comment]
Hi together,

I am Johanna Kleinekorte from Evonik, Germany. As I already heard about Idaes during my Phd at RWTH Aachen University, I now try to implement this impressive package as tool in the chemical industry for solving superstructure optimization problems. However, I am missing some published features such as pyosyn (https://doi.org/10.1016/j.compchemeng.2021.107414) in the current repository. Do I just have to be patient as it hasn't found its way into the public repo yet or am I missing something?

Thanks in advance and great work!
Best, Johanna

**Comment by andrewlee94** [comment]
Hi Johanna,

First of all, welcome and thank you for trying out our tools.

Some of the tools/features we have developed have been incorporated directly in the Pyomo code base so you might want to look there as well. I believe Pysosyn was one of these which is now part of Pyomo’s general GDPOpt package.

Otherwise, if you could tell us which tools or features you are interested in, we could help point you to where they now reside. We have reorganized our code base a few times, so links from old papers may no longer work unfortunately.

Andrew

From: Johanna Kleinekorte ***@***.***>
Sent: Tuesday, September 6, 2022 1:22 PM
To: IDAES/idaes-pse ***@***.***>
Cc: Lee, Andrew (FN) ***@***.***>; Comment ***@***.***>
Subject: [EXTERNAL] Re: [IDAES/idaes-pse] Welcome to idaes-pse Discussions! (Discussion #116)


Hi together,

I am Johanna Kleinekorte from Evonik, Germany. As I already heard about Idaes during my Phd at RWTH Aachen University, I now try to implement this impressive package as tool in the chemical industry for solving superstructure optimization problems. However, I am missing some published features such as pyosyn (https://doi.org/10.1016/j.compchemeng.2021.107414) in the current repository. Do I just have to be patient as it hasn't found its way into the public repo yet or am I missing something?

Thanks in advance and great work!
Best, Johanna

—
Reply to this email directly, view it on GitHub<https://github.com/IDAES/idaes-pse/discussions/116#discussioncomment-3574721>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AD4VSXY2DUFU6DP5K4B6X7TV454SNANCNFSM4USUDZEQ>.
You are receiving this because you commented.Message ID: ***@***.******@***.***>>
********************************************************************
This message does not originate from a known Department of Energy email system.
Use caution if this message contains attachments, links or requests for information.

********************************************************************
